### ISSUE TITLE
Enable and fix/baseline worthwhile FXCop rules on CoreFX codebase

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#14095]
dotnet/coreclr#7880 covers this for mscorlib.

@alexperovich will drive this.

Suggested workflow
* Put checkboxes in this issue for all valuable rules (not any that are too noisy/not valuable)
* In one commit, enable a rule, fix any easy violations and baseline the rest. This prevents us going backwards.
* Open separate issue for someone to clear out the baseline for the rule/s.

**Comment by bartdesmet** [comment]
Related to this, I've started to clean up some suppressions in `System.Linq.Expressions` that are no longer relevant, see https://github.com/dotnet/corefx/pull/14092. What's the recommended / well-supported way to run FxCop on .NET Core binaries? I couldn't find a note describing the steps in the Wiki.

**Comment by danmosemsft** [comment]
@alexperovich can answer that

**Comment by alexperovich** [comment]
FxCop the executable doesn't work on .NET Core binaries. We are running the roslyn fxcop analyzers in corefx.

**Comment by danmosemsft** [comment]
I think @bartdesmet may be asking for an example of using the roslyn analyzers.

**Comment by alexperovich** [comment]
The analyzers are run as part of build in any corefx source project using https://github.com/dotnet/corefx/blob/master/CodeAnalysis.ruleset as the ruleset. If you add nuget references to analyzer packages in project.jsons then they will also be included.

**Comment by alexperovich** [comment]
The compiler warnings are already enabled. CA2002 enabled with #14289

**Comment by tarekgh** [comment]
@alexperovich what else need to be done here?

**Comment by alexperovich** [comment]
The analyzer package is not completely implemented yet. I am waiting on roslyn to finish implementing the remaining analyzers.
